tekton/resources/nightly-release/overlays/pipeline/template.yaml

```diff
@@ -0,0 +1,57 @@
+- op: add
+  path: "/spec/resourcetemplates/-"
+  value:
+    apiVersion: tekton.dev/v1alpha1
+    kind: PipelineRun
+    metadata:
+      generateName: pipeline-release-nightly-$(uid)-
+    spec:
+      pipelineRef:
+        name: pipeline-release-nightly
+      params:
+      - name: package
+        value: $(params.gitrepository)
+      - name: imageRegistry
+        value: gcr.io/tekton-nightly
+      - name: versionTag
+        value: $(params.versionTag)
+      resources:
+      - name: source-repo
+        resourceRef:
+          name: git-source-$(uid)
+      - name: bucket
+        resourceRef:
+          name: tekton-bucket-nightly-$(uid)
+      - name: builtBaseImage
```

Comment:
We don't really benefit from using the image resource here - I wonder if we should just switch to an array of targets. Or perhaps we could support a new input type "list of resources" :)

Reply:
This is just copying the existing behavior: https://github.com/tektoncd/plumbing/blob/e34ec9c0a73900e953b726d93343c61e462bb4eb/tekton/resources/pipeline-nightly-release/pipeline-nightly-build-trigger.yaml#L96-L135

We need to at the moment because of the pipeline Pipeline: https://github.com/tektoncd/pipeline/blob/5af2878a30cf7a65e97bfe052bd00dd2a7133c09/tekton/release-pipeline-nightly.yaml#L13-L37.

Reply:
Yes, absolutely, this was just a general comment on things we might do in future, not for this patch for sure :)

Reply:
I tend to agree with you @afrittoli, but yeah, this is something to think about for later :angel: